### DISCUSSION TITLE
add new words to the dictionary on the fly

### DISCUSSION THREAD

**Discussion by Anuradhaks** [wolfgarbe/symspell#113] [Q&A]
1. Is there  a way to add new words to the existing dictionary on the fly without having to reload the dictionary  in c#

**Comment by wolfgarbe** [comment]
Yes, with CreateDictionaryEntry(string key, Int64 count);
key ist the word in lower case
count is the word frequency, used for ranking of suggestions

**Reply by Anuradhaks** [reply]
Thanks for the reply , once we have added the word, can we export/merge it to the existing dictionary file and save it on exit

**Reply by wolfgarbe** [reply]
There is no SaveDictionary method in SymSpell. An new word that you are creating with CreateDictionaryEntry lives only in RAM as long as the program is running.

But of course you can append a new word to the dictionary file yourself, e.g. with File.AppendAllText(dictionaryPath, "MyNewWord 1000");
